### PR TITLE
[redesign][ln] Receive tab

### PR DESCRIPTION
**Pull request by bgptr**
Toward issue #3296

Some notes:
- show settle date on invoice modal instead of expiration time when the payment already has been received
- cancel invoice feature added
- add snackbar messages for indicating add and cancel invoice events
- implemented filter, sort, and search function for invoice list
- tests added

Updated screenshots:
DeepinScreenshot_select-area_20210809215032
DeepinScreenshot_select-area_20210805140422
DeepinScreenshot_select-area_20210809214956

### CODE REVIEW

**Review by matheusd — COMMENTED** [review]
This is going in the right track, but I see some UX issues:

- You can't create a 0 amount invoice anymore. This is needed for "donations" (a 0 amount invoice indicates the sender can send any amount)
- Both canceled and timed out invoices are appearing as "canceled" (not sure that's the intention)
- Maybe we should disable copying the payment request if the payment was canceled/timed out? It will fail if someone attempts to pay for it, so this could make it less prone to cause this error
- Amount in the individual invoice box (in the invoices listing) is showing a "+" sign
- I'd like to get an additional expandable view (either in the details modal or directly in the listing) with the extended info for the invoice (hash, description, htlcs, etc).

**Comment by bgptr** [comment]
@matheusd Thanks for your review. I've pushed the suggested changes and updated the screenshots.

**Comment by bgptr** [comment]
Update: implemented the polished details part of the invoice modal thanks to @MariaPleshkova. (screenshots are updated)
Also, converted the details table to a shared component since more of the ln pages will use it.

File: app/components/modals/LNInvoiceModal/helpers.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { FormattedMessage as T } from "react-intl";
+export const getInvoiceDetails = (invoice, tsDate) => {
+  const details = [
+    {
+      label: <T id="ln.invoicesModal.hash" m="Hash" />,
+      value: invoice?.rHash
```
Comment (matheusd) [inline app/components/modals/LNInvoiceModal/helpers.js:6]: This is being displayed as a base64 encoding. Usually this is shown as an hex.

**Review by matheusd — COMMENTED** [review]
Also, it's probably useful to have the payment hash be copyable, since that's used in the command line to verify things.

**Comment by MariaPleshkova** [comment]
@bgptr 
I added the copy button next to the hash as @matheusd suggested. We already use the same button for the Remote Pubkey in the Channel Properties.
image

**Comment by bgptr** [comment]
@matheusd @MariaPleshkova Thanks, guys. I pushed the changes and updated the screenshots.

**Review by matheusd — APPROVED** [review]
Working nicely now!

**Pull request merged**